dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

serjiokov opened the pull request:
#15951 UI unresponsive while loading a larger JSONB

Provided functionality for restriction of large file content representation.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB
object

Provided functionality for restriction of large file content
representation.

E1izabeth commented:
Please use 4 spaces instead of tabs for indentation

E1izabeth commented:
Could you also provide a screenshot of how does it look like?

E1izabeth commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -271,42 +289,66 @@ public <T> T getAdapter(Class<T> adapter) {
+				return;
+			}
+			resetEditorInput();
+			byte[] bytes = CommonUtils.toString(value).getBytes();
```
Why don't you use getContentLength method?

serjiokov replied:
The main reason is in next: funtion "getContentLength" packed to ContentEditorInput here such object is unaccessable, (another logic branch is for the case of limited content Im using StringEditorInput object) that is why using getBytes is applicable here.

E1izabeth replied:
I referred to `DBDContent`'s method as value is a `DBDContent`

serjiokov replied:
ok, will do.

E1izabeth commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -271,42 +289,66 @@ public <T> T getAdapter(Class<T> adapter) {
+		});
+	}
+
+	private void showRegularContent(final PanelEditorPaintListener editorPaintListener, byte[] bytes,
+			DBRProgressMonitor monitor) throws DBException {
```
We use such formatting for a long lines
```suggestion
	private void showRegularContent(
	    @NotNull final PanelEditorPaintListener editorPaintListener,
	    byte[] bytes,
	    @NotNull DBRProgressMonitor monitor
	) throws DBException {
```

serjiokov replied:
Let's have agrement of code style, Im currently  using default eclipse settings.

E1izabeth replied:
We have it already.

Such formatting for long lines
Long line is 140 symbols and more
Formatting for casts is the following `(ClassName) varName` - space in between
Arrays formatting: `new String[]{"*.*", "*.jks", "*"}`
Don't use static imports
Also note that we use nullable annotations.

E1izabeth replied:
Also we don't write if or cycles without blocks. For example:
```
if (true) System.out.println("Hey!");
```
should be 
```
if (true) {
    System.out.println("Hey!");
}
```

serjiokov replied:
It is clear, same for each IDE,
one more point, the max line width is 140  instead of 120 (120 by defaul) is it correct?

serjiokov commented:
The current behavior presented on the screen:
![image](https://github.com/dbeaver/dbeaver/assets/32031248/b797392a-3564-4231-b2aa-3fd6c726ee3f)

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed code style 4 spaces instead of tabs,etc

E1izabeth commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -269,44 +287,71 @@ public <T> T getAdapter(Class<T> adapter) {
-                UIUtils.asyncExec(() -> {
-                    editor.setInput(new StringEditorInput("Empty", "", true, StandardCharsets.UTF_8.name()));
-                });
+            int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore()
+                    .getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES) * 1000;
```
Also we don't use 8 spaces for indentation in such places. Always 4 spaces :)
```suggestion
            int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore()
                .getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES) * 1000;
```

serjiokov replied:
Already reformatted

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Reworked to use getContentLength from DBValue object

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Applied codestyle for UIUtils

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed code style PrefPageResultSetMain

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Code style check fixed new method in  UIUtils class

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed Javadoc tag @param should be preceded with an empty line.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed Javadoc tag @param should be preceded with an empty line.

serge-rider commented on plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/UIUtils.java:
```diff
@@ -1512,6 +1512,26 @@ public static void drawMessageOverControl(Control control, GC gc, String message
+     * @param gc      - graphical container
+     * @param message - custom message text
+     */
+    public static void drawMessageOverControlOnLeftBottom(Control control, GC gc, String message) {
```
Let's leave this method in panel editor. UIUtils is already too big and this implementation looks quite specific.

serjiokov replied:
The drawing method content moved to PanelEditorPaintListener.

serge-rider commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -269,44 +287,69 @@ public <T> T getAdapter(Class<T> adapter) {
+            } else {
+                showRegularContent(monitor);
             }
-            UIUtils.asyncExec(() -> {
```
Why default progress message was removed? It still makes sense I think.

serjiokov replied:
We added content size restriction, the large values do not take much loading time now.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Drawing function moved to paintListener in AbstractTextPanelEditor

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Code style check in AbstractTextPanelEditor

MashaKorax commented:
If the value of **Maximum size of editing content** was set =< two-digit number (xx)---> after that it is impossible to change it to more than two digits. The same with 1 digit.

https://github.com/dbeaver/dbeaver/assets/84867187/1ab27cb7-147d-403f-b792-11465692aaa6

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed layout of groups controls to avoid collapsing text editors.
Minimal size added also.

serjiokov commented:
![image](https://github.com/dbeaver/dbeaver/assets/32031248/f5f0e717-b0ec-4761-b30b-091dafbb9718)
Aligned ui components

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Code style fixed

MashaKorax committed:
Merge branch 'devel' into bug/#15951-jsonb-query-freezing

MashaKorax commented:
additional tiket https://github.com/dbeaver/pro/issues/1952

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -107,7 +118,15 @@ public StyledText createControl(IValueController valueController) {
         editor.addContextMenuContributor(manager -> contributeTextEditorActions(manager, editorControl));
+        editorPaintListener = new PanelEditorPaintListener();
+        editorControl.addPaintListener(editorPaintListener);
+        editorControl.addListener(SWT.RESIZE, new Listener() {
```
Wrong constant

```suggestion
        editorControl.addListener(SWT.Resize, new Listener() {
```

serjiokov replied:
```
/**
	 * Style constant for resize box trim (value is 1&lt;&lt;4).
	 * <p><b>Used By:</b></p>
	 * <ul>
	 * <li><code>Decorations</code> and subclasses</li>
	 * <li><code>Tracker</code></li>
	 * </ul>
	 */
	public static final int RESIZE = 1 << 4;
```

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -451,4 +497,34 @@ public String getEncoding() {
         }
     }
+    
+    class PanelEditorPaintListener implements PaintListener {
+
+        private String statusMessage;
+
+        public PanelEditorPaintListener() {
+            this.statusMessage = "";
+        }
+
+        public void updateStatusMessage(String statusMessage) {
+            this.statusMessage = statusMessage;
+        }
+
+        public void paintControl(PaintEvent event) {
+            Object source = event.getSource();
+            if (source instanceof Control) {
+                GC gc = event.gc;
+                Rectangle bounds = ((Control) source).getBounds();
+                gc.setForeground(UIUtils.getSharedColor(SharedTextColors.COLOR_WARNING));
+                final int height = gc.textExtent(statusMessage).y;
+                for (String line : statusMessage.split("\n")) {
+                    line = line.trim();
+                    Point ext = gc.textExtent(line);
+                    int x = bounds.x;
+                    int y = bounds.y + bounds.height - ext.y - height;
+                    gc.drawText(line, x, y);
+                }
+            }
+        }
+    }
```
Why do we need to paint text over a control? Can't we just create a new label control?

serjiokov replied:
No, it is drawn directly on canvas, here require a specific text as a message. The label require composite.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/editors/data/preferences/PrefPageResultSetMain.java:
```diff
@@ -123,127 +127,149 @@ protected boolean supportsDataSourceSpecificOptions()
+        rightPane.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+        // Group of queries settings
+        Group queriesGroup = UIUtils.createControlGroup(leftPane,
```
Please bring back {}. They denote a scope. You don't need the `queriesGroup` variable outside. Same for blocks down below.

serjiokov replied:
{} - is redundunt and does not fit the Java Code Style convention.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/editors/data/preferences/PrefPageResultSetMain.java:
```diff
@@ -293,7 +319,9 @@ protected void loadPreferences(DBPPreferenceStore store)
             showErrorsInDialog.setSelection(store.getBoolean(ResultSetPreferences.RESULT_SET_SHOW_ERRORS_IN_DIALOG));
             markCellValueOccurrences.setSelection(store.getBoolean(ResultSetPreferences.RESULT_SET_MARK_CELL_VALUE_OCCURRENCES));
-
+            int maxEditiingContentSize = store.getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES);
+            maxEditingContentSize.setText(String.valueOf(maxEditiingContentSize));
```
Redundant local variable

```suggestion
maxEditingContentSize.setText(String.valueOf(store.getInt(ModelPreferences.EDITING_CONTENT_MAX_SIZE_KBYTES)));
```

serjiokov replied:
The variable extracted for clarity.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Constant aggregated to function, Resize used.

serjiokov committed:
Merge branch 'bug/#15951-jsonb-query-freezing' of github.com:dbeaver/dbeaver into bug/#15951-jsonb-query-freezing

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Rework to message label as a status panel on the bottom editor.
Revert preferences page code for better day
Restricted range of representing content value.

serjiokov commented:
Rework to message label panel text in place of editor.
![Screenshot 2023-09-19 003745](https://github.com/dbeaver/dbeaver/assets/32031248/6d35bc21-b136-43c8-85de-9430fe2f1e5a)

serjiokov commented:
New group added for detup represention content on prefernces page.
![Screenshot 2023-09-19 014610](https://github.com/dbeaver/dbeaver/assets/32031248/93ca5d83-0d23-42b8-a230-ad1fa134af90)

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Code style fixed

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Code style fixed
throws has incorrect indentation level 8, expected level should be 12

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/internal/ResultSetMessages.properties:
```diff
@@ -245,6 +245,7 @@ pref_page_content_editor_checkbox_use_navigator_filters_tip = Filter (hide) colu
 pref_page_content_editor_group_content = LOB
 pref_page_content_editor_group_hex = Binary editor
 pref_page_content_editor_group_editing = Behavior on missing key
+pref_page_content_editor_group_representation_content = Representation content
```
I'm not a native speaker, but I think `Content representation` sounds better. What do you think?

serjiokov replied:
I think so, thanks.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -101,16 +115,44 @@ public StyledText createControl(IValueController valueController) {
 
+    private void reasignLayout(Composite cmpsInternalBase) {
+        Control[] children = cmpsInternalBase.getChildren();
+        Stream.of(children).forEach(c -> c.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1)));
```
Just use plain foreach loop, no need to use streams here. You can also reuse the created layout.

Also, I'm not even sure why this is needed. You can nest several composites to avoid changing child layout data. It looks hacky.

serjiokov replied:
So, the nested ui controls out of controls and layouting here, the default class is pure Eclipse implementaion that hided in abstract root class with Stack layout, whitout this approch Label looks ugly.

serjiokov replied:
...mmm not get you right about streams, whats is you consern?

ShadelessFox replied:
I've explained my concern about streams in another comment.

Regarding layout, you can wrap the editor's component in another composite and change its layout instead.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -101,16 +115,44 @@ public StyledText createControl(IValueController valueController) {
+    private CLabel createMessageBar(Composite cmpsBase) {
+        UIUtils.createHorizontalLine(cmpsBase);
+        messageBar = new CLabel(cmpsBase, SWT.RIGHT);
+        GridData layoutData = new GridData(SWT.FILL, SWT.FILL, true, false, 1, 1);
```
Redundant local variable

serjiokov replied:
Reorganized

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -269,28 +311,64 @@ public <T> T getAdapter(Class<T> adapter) {
     @Override
-    public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull StyledText control, @Nullable DBDContent value) throws DBException
-    {
+    public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull StyledText control, @Nullable DBDContent value)
+            throws DBException {
```
```suggestion
    public void primeEditorValue(
        @NotNull DBRProgressMonitor monitor,
        @NotNull StyledText control,
        @Nullable DBDContent value
    ) throws DBException {
```

serjiokov replied:
Redundant style - supported only by IDEA. Jenkins style check past. No issue here.

serjiokov replied:
Meanwhile All checks have passed
4 successful checks

ShadelessFox replied:
Then use IDEA instead. We use this style of wrapping for long methods.

serjiokov replied:
Meanwhile All checks have passed
4 successful checks

ShadelessFox replied:
It's not about checks - they are dumb anyway - it's about human readability.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -299,12 +377,25 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
+        int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore()
+            .getInt(ModelPreferences.REPRESENTATION_CONTENT_MAX_SIZE_KBYTES);
+        byte[] bytes = CommonUtils.toString(value).getBytes();
+        byte[] restrictedBytes = Arrays.copyOfRange(bytes, 0, maxContentSize);
```
You should use `Math.min(bytes.length, maxContentSize)` to avoid having zero bytes at the end if the buffer is smaller than `maxContentSize`.

serjiokov replied:
What issues with zero at the end? The restricted content is cutting and is in read only mode, auto formating aslo does not work for this case.

ShadelessFox replied:
If the provided content is shorter than the limit, the resulting buffer will be filled with zero bytes to be of size `maxContentSize`.

serjiokov replied:
Should not happen because of 
```
if (value.getContentLength() > maxContentSize) {
                showRestrictedContent(value);
            } else {
                showRegularContent(monitor);
            }
```

ShadelessFox replied:
But it's happening. Also content conversion is off in SQLite (it should display trimmed data, not the displayed placeholder):

|Without my suggestion|With my suggestion|
|---|---|
|![java_mpO8C4cO13](https://github.com/dbeaver/dbeaver/assets/35821147/2936ddd5-a40e-4b0a-9dc3-27c170f6db1e)|![ZZPAk7FhiL](https://github.com/dbeaver/dbeaver/assets/35821147/72f5ca95-b7af-4951-a73d-e470182e54a2)|

serjiokov replied:
Content restricted by 1000 KB, but the right part looks a 100 bytes, not sure is it correct.

ShadelessFox replied:
You're basically doing `content.toString()`, which is incorrect. You should use `DBDContent#getContents` instead.

ShadelessFox replied:
This is a bug, and I won't pass this PR to the QA team until it's fixed.

E1izabeth replied:
Let's better use DBDContent#getContents. No need to call toString when we have API for that needs.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -451,4 +542,34 @@ public String getEncoding() {
         }
     }
+    
+    class PanelEditorPaintListener implements PaintListener {
```
Unused code.

serjiokov replied:
Unused code removed.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Fixed message text

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Unused class removed.

serjiokov committed:
dbeaver/dbeaver#15951 UI unresponsive while loading a larger JSONB

Unused variable removed.

ShadelessFox reviewed (commented):
Please see my other comments

serjiokov commented:
Please argue requested style  change if all checks have been passed (4 successful checks)
Please respond on my comments.

serjiokov committed:
Reorganized method signature to column arg style

serjiokov committed:
dbeaver/dbeaver#15951 Reorganized to read values from DBDContentStorage

serjiokov committed:
dbeaver/dbeaver#1017 Reorganized to read values from DBDContentStorage

serjiokov committed:
Merge branch 'bug/#15951-jsonb-query-freezing' of github.com:dbeaver/dbeaver into bug/#15951-jsonb-query-freezing

Matvey16 commented:
Inputing anything feels very odd, because any number inputed immidiately turns to `100` and other input messes up.


https://github.com/dbeaver/dbeaver/assets/82543000/9f6e48b6-d890-4ab5-934d-9a7a26631fd5

serjiokov committed:
dbeaver/dbeaver#15951 Removed limits (input key listener)

Matvey16 commented:
Verified

E1izabeth committed:
Merge branch 'devel' into bug/#15951-jsonb-query-freezing

serge-rider commented:
What confuses me:
- New option on main data editor pref page. It should be on Apperance or on Binary Editor page
- Existing option "Maximum text editor content length" look very similar. And it also looks like we don't use it anymore. Let's delete it?

serjiokov commented:
I can propose reuse already existing pref option just rename a group name:
![image](https://github.com/dbeaver/dbeaver/assets/32031248/2776c761-014f-447d-abb3-8dae332b25ff)
PLS give your feedback

serjiokov committed:
dbeaver/dbeaver#15951 Reorganized preferences page for Max Content
Length

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/editors/data/preferences/PrefPageResultSetMain.java:
```diff
@@ -214,36 +215,25 @@ public void widgetSelected(SelectionEvent e) {
-            });
-        }
-
+        GridLayout editingGroupLayout = new GridLayout(2, false);
+        final Group group = UIUtils.createControlGroup(leftPane,
+                ResultSetMessages.pref_page_content_editor_group_editing, 1, GridData.VERTICAL_ALIGN_BEGINNING, 0);
+        group.setLayout(editingGroupLayout);
+
+        alwaysUseAllColumns = UIUtils.createCheckbox(group,
+                ResultSetMessages.pref_page_content_editor_checkbox_keys_always_use_all_columns, false);
+        alwaysUseAllColumns.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true, 2, 1));
+
+        disableEditingOnMissingKey = UIUtils.createCheckbox(group,
+                ResultSetMessages.pref_page_content_editor_checkbox_disable_editing_if_key_missing, false);
+        disableEditingOnMissingKey.setLayoutData(new GridData(GridData.FILL, GridData.FILL, true, true, 2, 1));
```
Please revert this redundant change

serjiokov replied:
reverted

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -299,11 +377,27 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
+        int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ResultSetPreferences.RS_EDIT_MAX_TEXT_SIZE);
+        DBDContentStorage contents = value.getContents(new VoidProgressMonitor());
+        try (final InputStream stream = contents.getContentStream()) {
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream((int) contents.getContentLength());
+            IOUtils.copyStream(stream, buffer);
+            byte[] displaingContentBytes = Arrays.copyOfRange(buffer.toByteArray(), 0, maxContentSize * 1000);
```
Can we read `maxContentSize` instead of the entire blob first and then slice it? It might be memory-expensive and won't fit in a 4GB boundary for big BLOBs.

serjiokov replied:
Yeap sure, let rework to read require content lenght.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -299,11 +377,27 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
+        });
+    }
+
+    private void showRestrictedContent(@NotNull DBDContent value) throws DBCException, IOException {
```
Let's also rename it as "restricted" doesn't quite fit the purpose of this method. Let's call it something like "showLimitedContent", you get the spirit.

Also, let's pass `maxContentSize` as an argument (already computed by the caller) to avoid code duplication.

serjiokov replied:
Done

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -269,28 +310,65 @@ public <T> T getAdapter(Class<T> adapter) {
+                            UIUtils.drawMessageOverControl(textWidget, gc,
+                                NLS.bind(ResultSetMessages.panel_editor_text_loading_placeholder_label, textInput.getContentLength()), 0);
                             editor.setInput(textInput);
+                            messageBar.setForeground(UIStyles.getDefaultTextForeground());
+                            messageBar.setText(String.format("Content size: %s Bytes", textInput.getContentLength()));
```
I'm unsure whether this label should be always displayed or not. It's probably up to a discussion.

serjiokov replied:
Persistant panel looks empty and ugly without information, displaing content lenght is kind of interesting to have. But let's enchance it after UI/UX discussion.

serjiokov committed:
dbeaver/dbeaver#15951 Renamed methods, load chunk of data

ShadelessFox commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java:
```diff
@@ -381,16 +381,15 @@ private void showRegularContent(@NotNull DBRProgressMonitor monitor) throws DBEx
-            IOUtils.copyStream(stream, buffer);
-            byte[] displaingContentBytes = Arrays.copyOfRange(buffer.toByteArray(), 0, maxContentSize * 1000);
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream(lengthInBytes);
+            IOUtils.copyStream(stream, buffer, lengthInBytes);
```
The third argument is the internal `bufferSize`, not the maximum amount of bytes to read.

Just create a byte array and use `stream.read(buffer)` to read it, it will return the length of bytes read which you can use to trim the output array if fewer bytes were read.

serjiokov replied:
Better to use pure readNBytes. 
`byte[] displaingContentBytes = stream.readNBytes(lengthInBytes);`

serjiokov committed:
dbeaver/dbeaver#15951 Limit size of reading DataContent

Matvey16 commented:
Verified
